### PR TITLE
[IIIF-645][IIIF-771] Continuous Deployment and YAML-->Terraform Kubernetes Deployment

### PR DESCRIPTION
**Pull request by cachemeoutside**
This is a really large PR as we went over it today.

The test environment is live with this new setup.
https://test-iiif.library.ucla.edu/iiif/2/healthcheckimage/full/full/0/default.jpg
https://test-iiif.library.ucla.edu/fester/docs

The Jenkins job is defined here. It can be triggered via TravisCI. But that'll be scoped in a new ticket. As of now, manual deploys are trigger-able.
https://jenkins-devsupport.library.ucla.edu/job/kubernetes_iiif_deployment/

The PR consists of the following changes:
* CodeBuild managed by Terraform
* IAM policies managed by Terraform
* Kubernetes resources and deployment managed by Terraform
* .buildspec.yaml file added to connect Kubernetes deployment to CodeBuild
* Jenkinsfile added as trigger to CodeBuild and also serve as a manual deployment interface

### CODE REVIEW

**Comment by cachemeoutside** [comment]
clean up of old directories will be completed in IIIF-776

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+pipeline {
+    agent any
+    stages {
+      stage("Deploying to EKS") {
+        parallel {
+          stage("Deploy via AWS CodeBuild") {
+            steps {
+              awsCodeBuild(
+              projectName: "iiif-k8s-deployment",
```
Comment (ksclarke) [inline Jenkinsfile:9]: No indentation for the things inside the parens?

File: environments/codebuild/remote-state.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+### Retrieve IAM arns to use
+data "terraform_remote_state" "iam" {
+  backend = "s3"
+  config = {
+    bucket = "prod-services-terraform-state"
```
Comment (ksclarke) [inline environments/codebuild/remote-state.tf:5]: So, like, here has unaligned equals.

File: environments/iam/k8s.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Create a default k8s robot role account for k8s deployments
+resource "aws_iam_role" "k8s_robot" {
+  name               = "${var.prefix_tag}_${var.eks_deployment_robot_name}"
```
Comment (ksclarke) [inline environments/iam/k8s.tf:3]: and here the equals are aligned. Would be nice to be consistent: one style or the other. Personally I don't like the equals aligned, but it's up to you.

Comment (cachemeoutside) [reply environments/iam/k8s.tf]: Will create a ticket to research ideal formatting practice. Although I think running `terraform fmt` conforms to the preferred style and would make it easier to format all terraform configs. But better to do some research first.

Comment (cachemeoutside) [reply environments/iam/k8s.tf]: In the meantime, I've reverted this to be unaligned

File: Jenkinsfile (hunk 1 above)
Comment (cachemeoutside) [reply Jenkinsfile]: fixed indentation and spacing in new push.

**Pull request merged**